dist/tools/generate_c11_atomics_cpp_compat_header/generate_c11_atomics_cpp_compat_header.sh

```diff
@@ -0,0 +1,134 @@
+#!/bin/sh
+
+TESTFILE=/tmp/riot_are_types_of_same_size_compilation_check.c
+
+ATOMIC_TYPES="
+    atomic_bool
+    atomic_char
+    atomic_schar
+    atomic_uchar
+    atomic_short
+    atomic_ushort
+    atomic_int
+    atomic_uint
+    atomic_long
+    atomic_ulong
+    atomic_llong
+    atomic_ullong
+    atomic_int_least8_t
+    atomic_uint_least8_t
+    atomic_int_least16_t
+    atomic_uint_least16_t
+    atomic_int_least32_t
+    atomic_uint_least32_t
+    atomic_int_least64_t
+    atomic_uint_least64_t
+    atomic_int_fast8_t
+    atomic_uint_fast8_t
+    atomic_int_fast16_t
+    atomic_uint_fast16_t
+    atomic_int_fast32_t
+    atomic_uint_fast32_t
+    atomic_int_fast64_t
+    atomic_uint_fast64_t
+    atomic_intptr_t
+    atomic_uintptr_t
+    atomic_size_t
+    atomic_ptrdiff_t
+    atomic_intmax_t
+    atomic_uintmax_t
+    "
+
+are_types_of_same_size() {
+    cat > $TESTFILE << EOF
+#include <stdatomic.h>
+#include <stdint.h>
+int main(void) {
+    _Static_assert(sizeof($1) == sizeof($2));
```

Comment:
```suggestion
    _Static_assert(sizeof($1) == sizeof($2), "size does not match");
```
`_Static_assert` allow a message parameter. On ESP32 with gcc 5.2.0, it obviously requires to message parameter to work.